monitor.cpp

```diff
@@ -1,38 +1,49 @@
 #include "./monitor.h"
-#include <assert.h>
 #include <thread>
 #include <chrono>
 #include <iostream>
 using std::cout, std::flush, std::this_thread::sleep_for, std::chrono::seconds;
 
-int vitalsOk(float temperature, float pulseRate, float spo2) {
-  if (temperature > 102 || temperature < 95) {
-    cout << "Temperature is critical!\n";
+// Pure functions
+bool isTemperatureOk(float temperature) {
+    return (temperature >= 95 && temperature <= 102);
+}
+bool isPulseOk(float pulseRate) {
+    return (pulseRate >= 60 && pulseRate <= 100);
+}
+bool isSpO2Ok(float spo2) {
+    return (spo2 >= 90);
+}
+
+// I/O function
+void blinkAlert(const char* message) {
+#ifdef UNIT_TEST
+    (void)message;
+#else
+    cout << message << "\n";
     for (int i = 0; i < 6; i++) {
-      cout << "\r* " << flush;
-      sleep_for(seconds(1));
-      cout << "\r *" << flush;
-      sleep_for(seconds(1));
+        cout << "\r* " << flush;
+        sleep_for(seconds(1));
+        cout << "\r *" << flush;
+        sleep_for(seconds(1));
     }
-    return 0;
-  } else if (pulseRate < 60 || pulseRate > 100) {
-    cout << "Pulse Rate is out of range!\n";
-    for (int i = 0; i < 6; i++) {
-      cout << "\r* " << flush;
-      sleep_for(seconds(1));
-      cout << "\r *" << flush;
-      sleep_for(seconds(1));
+    cout << "\r  \r" << flush;
+#endif
+}
+
+// Main function
+int vitalsOk(float temperature, float pulseRate, float spo2) {
```

Comment:
The `vitalsOk` function still has a complexity of `4`. By itself, that's not a big number. But as newer vitals get added, its complexity keeps increasing. Notice a small repetition for each vital - `if`, followed by the vital-specific function, followed by `blinkAlert`, then `return 0`.
If you factor out the repetitions and make a function with parameters, then you could check all vitals in a loop. Try it out